src/librustc/session/mod.rs

```diff
@@ -1138,7 +1134,11 @@ fn build_session_(
     driver_lint_caps: FxHashMap<lint::LintId, lint::Level>,
 ) -> Session {
     let self_profiler =
-        if sopts.debugging_opts.self_profile { Some(Arc::new(PlMutex::new(SelfProfiler::new()))) }
+        if sopts.debugging_opts.self_profile {
+            let profiler = SelfProfiler::new();
```

Comment:
Once this uses proper error handling, I think failing to create the profiler should not be a hard error. Emitting a warning and setting `self_profiler` to `None` should provide the better user experience.